src/Parsers.jl

```diff
@@ -215,6 +215,28 @@ Return `Pair(field-name => field-value)` and
 a `SubString` containing the remaining header-field lines.
 """
 function parse_header_field(bytes::SubString{String})::Tuple{Header,SubString{String}}
+    # https://github.com/JuliaWeb/HTTP.jl/issues/796
+    # there may be certain scenarios where non-ascii characters are
+    # included (illegally) in the headers; curl warns on these
+    # "malformed headers" and ignores them. we attempt to re-encode
+    # these from latin-1 => utf-8 and then try to parse.
+    if !isvalid(bytes)
+        @warn "malformed HTTP header detected; attempting to re-encode from Latin-1 to UTF8"
+        rawbytes = codeunits(bytes)
+        buf = Base.StringVector(length(rawbytes) + count(≥(0x80), rawbytes))
```

Comment:
Should we add a fallback for when `Base.StringVector` does not exist?

Reply:
We're requiring Julia 1.6 in HTTP.jl now, so I think we should be covered?

Reply:
It's just that this is not public Julia API (it has no docstring). But I guess that if Stefan recommends it then it should be fine?